bentoml/_internal/utils/analytics/schemas.py

```diff
@@ -0,0 +1,223 @@
+import os
+import re
+import hmac
+import uuid
+import typing as t
+import hashlib
+from abc import ABC
+from typing import TYPE_CHECKING
+from datetime import datetime
+from datetime import timezone
+from platform import platform
+from platform import python_version
+from functools import lru_cache
+
+import yaml
+import cattr
+import psutil
+import attrs.converters
+from simple_di import inject
+from simple_di import Provide
+
+from ...configuration import BENTOML_VERSION
+from ...configuration.containers import BentoMLContainer
+from ...yatai_rest_api_client.config import get_config_path
+from ...yatai_rest_api_client.config import get_current_context
+
+if TYPE_CHECKING:
+    P = t.ParamSpec("P")
+    GenericFunction = t.Callable[P, t.Any]
+
+
+@lru_cache(maxsize=1)
+def get_platform() -> str:
+    return platform(aliased=True)
+
+
+@lru_cache(maxsize=1)
+def get_python_version() -> str:
+    return python_version()
+
+
+@attrs.define
+class ClientInfo:
+    client_id: str
+    client_creation_timestamp: datetime
+
+
+@inject
+@lru_cache(maxsize=1)
+def get_client_id(
+    bentoml_home=Provide[BentoMLContainer.bentoml_home],
+) -> t.Optional[ClientInfo]:
+    CLIENT_ID_PATH = os.path.join(bentoml_home, "client_id")
+
+    if os.path.exists(CLIENT_ID_PATH):
+        with open(CLIENT_ID_PATH, "r", encoding="utf-8") as f:
+            client_info = yaml.safe_load(f)
+        return ClientInfo(**client_info)
+    else:
+
+        def create_client_id() -> t.Dict[str, str]:
+            # returns an unique client_id and timestamp in ISO format
+            uniq = uuid.uuid1().bytes
+            client_id = hmac.new(uniq, digestmod=hashlib.blake2s).hexdigest()
```

Comment:
I don't think this is a big deal, but why are we doing this?

Reply:
I was learning about this earlier. We definitely don't have to do this